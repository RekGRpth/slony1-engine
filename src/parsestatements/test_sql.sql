select * from foo;  select * from bar;  select * from frobozz;
alter table foo add column c integer;
alter table foo alter column c set not null; 

-- Comment line that should hide a whole bunch of quoting... ;; $$
-- '"''"; "\"\"\"$$ ;"\"""

-- Here is an old-style pl/pgsql function using heavy quoting
create function foo (text) returns integer as '
  declare
     rc record;
  begin
    select * into rc from foo where name = ''Some Favored name'';
    return NULL;
  end;' language plpgsql;

select * from foo;  select * from bar;  select * from frobozz;

create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = 'Some Favored name';
    return NULL;
  end;
$$ language plpgsql;

select * from foo;  select * from bar;  select * from frobozz;

-- This isn't actually a particularly well-framed stored function
-- but it abuses $$dollar quoting$$ quite nicely...
create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = $23$Some Favored name$23$;
    -- Use a secondary bit of quoting to make sure that nesting works...
    select $24$ -- another " " thing ' ' \\\'\$ $24$;
    return NULL;
  end;
$$ language plpgsql;
