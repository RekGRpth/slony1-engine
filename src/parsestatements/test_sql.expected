Input: select * from foo;  select * from bar;  select * from frobozz;
alter table foo add column c integer;
alter table foo alter column c set not null; 

-- Comment line that should hide a whole bunch of quoting... ;; $$
-- '"''"; "\"\"\"$$ ;"\"""

-- Here is an old-style pl/pgsql function using heavy quoting
create function foo (text) returns integer as '
  declare
     rc record;
  begin
    select * into rc from foo where name = ''Some Favored name'';
    return NULL;
  end;' language plpgsql;

select * from foo;  select * from bar;  select * from frobozz;

create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = 'Some Favored name';
    return NULL;
  end;
$$ language plpgsql;

select * from foo;  select * from bar;  select * from frobozz;

-- This isn't actually a particularly well-framed stored function
-- but it abuses $$dollar quoting$$ quite nicely...
create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = $23$Some Favored name$23$;
    -- Use a secondary bit of quoting to make sure that nesting works...
    select $24$ -- another " " thing ' ' \\\'\$ $24$;
    return NULL;
  end;
$$ language plpgsql;


statement 0
-------------------------------------------
select * from foo;
statement 1
-------------------------------------------
  select * from bar;
statement 2
-------------------------------------------
  select * from frobozz;
statement 3
-------------------------------------------

alter table foo add column c integer;
statement 4
-------------------------------------------

alter table foo alter column c set not null;
statement 5
-------------------------------------------
 

-- Comment line that should hide a whole bunch of quoting... ;; $$
-- '"''"; "\"\"\"$$ ;"\"""

-- Here is an old-style pl/pgsql function using heavy quoting
create function foo (text) returns integer as '
  declare
     rc record;
  begin
    select * into rc from foo where name = ''Some Favored name'';
    return NULL;
  end;' language plpgsql;
statement 6
-------------------------------------------


select * from foo;
statement 7
-------------------------------------------
  select * from bar;
statement 8
-------------------------------------------
  select * from frobozz;
statement 9
-------------------------------------------


create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = 'Some Favored name';
    return NULL;
  end;
$$ language plpgsql;
statement 10
-------------------------------------------


select * from foo;
statement 11
-------------------------------------------
  select * from bar;
statement 12
-------------------------------------------
  select * from frobozz;
statement 13
-------------------------------------------


-- This isn't actually a particularly well-framed stored function
-- but it abuses $$dollar quoting$$ quite nicely...
create or replace function foo (text) returns integer as $$
  begin
    select * into rc from foo where name = $23$Some Favored name$23$;
    -- Use a secondary bit of quoting to make sure that nesting works...
    select $24$ -- another " " thing ' ' \\\'\$ $24$;
    return NULL;
  end;
$$ language plpgsql;